_data-prepper/log-analytics.md

```diff
@@ -0,0 +1,149 @@
+
+# Pipeline configuration
+
+## Example pipeline.yaml with SSL and basic authentication enabled
+
+Example `pipeline.yaml` with SSL and basic authentication enabled for the `http-source`:
+
+```yaml
+log-pipeline:
+  source:
+    http:
+      ssl_certificate_file: "/full/path/to/certfile.crt"
+      ssl_key_file: "/full/path/to/keyfile.key"
+      authentication:
+        http_basic:
+          username: "myuser"
+          password: "mys3cret"
+  processor:
+    - grok:
+        match:
+          # This will match logs with a "log" key against the COMMONAPACHELOG pattern (ex: { "log": "actual apache log..." } )
+          # You should change this to match what your logs look like. See the grok documenation to get started.
+          log: [ "%{COMMONAPACHELOG}" ]
+  sink:
+    - opensearch:
+        hosts: [ "https://localhost:9200" ]
+        # Change to your credentials
+        username: "admin"
+        password: "admin"
+        # Add a certificate file if you are accessing an OpenSearch cluster with a self-signed certificate  
+        #cert: /path/to/cert
+        # If you are connecting to an Amazon OpenSearch Service domain without
+        # Fine-Grained Access Control, enable these settings. Comment out the
+        # username and password above.
+        #aws_sigv4: true
+        #aws_region: us-east-1
+        # Since we are grok matching for apache logs, it makes sense to send them to an OpenSearch index named apache_logs.
+        # You should change this to correspond with how your OpenSearch indices are set up.
+        index: apache_logs
+```
+
+This pipeline configuration is an example of Apache log ingestion. Don't forget that you can easily configure the Grok processor for your own custom logs. You will need to modify the configuration above for your OpenSearch cluster.
+
+The main changes you need to make are:
+
+* `hosts` - Set to your hosts
+* `index` - Change this to the OpenSearch index you want to send logs to
+* `username`- Provide the OpenSearch username
+* `password` - Provide your OpenSearch password
+* `aws_sigv4` - If you use Amazon OpenSearch Service with AWS signing, set this to true. It will sign requests with the default AWS credentials provider.
+* `aws_region` - If you use Amazon OpenSearch Service with AWS signing, set this value to your region.
+
+## FluentBit
+
+You will have to run FluentBit in your service environment. You can find the installation guide of FluentBit [here](https://docs.fluentbit.io/manual/installation/getting-started-with-fluent-bit). Ensure that you can configure the [FluentBit http output plugin](https://docs.fluentbit.io/manual/pipeline/outputs/http) to your Data Prepper Http Source. Below is an example `fluent-bit.conf` that tails a log file named `test.log` and forwards it to a locally running Data Prepper's http source, which runs by default on port 2021. 
```

Comment:
Do not use _here_ or _click here_ for link text because it creates accessibility problems. In most cases, a formal cross-reference (the title of the page you're linking to) is the preferred style because it provides context and helps readers understand where they're going when they choose the link.

Is "configure...to" the right wording, or would "configure...for" or another preposition be better? Can we add the noun after `fluent-bit.conf`? Is "tails" the right verb, or can we use something a bit clearer?